tactics/hints.ml

```diff
@@ -664,12 +664,14 @@ struct
       (* Warn about no longer typable hint? *)
       None
 
+  exception NoHeadEvar
+
   let head_evar sigma c =
     let rec hrec c = match EConstr.kind sigma c with
       | Evar (evk,_)   -> evk
       | App (c,_)      -> hrec c
       | Cast (c,_,_)   -> hrec c
-      | _              -> raise Evarutil.NoHeadEvar
+      | _              -> raise NoHeadEvar
```

Comment:
Return an option type instead, this makes the code below much leaner.

Reply:
(Or actually just the boolean, it's only used once.)

Reply:
OK, will do.

Reply:
@herbelin there is still this little nitpick.